### DISCUSSION TITLE
Fixing weather_HassGetWeather and person_HassGetState

### DISCUSSION THREAD

**Discussion by tetele** [OHF-Voice/intents#1768] [Supported Sentences]
Hey @home-assistant/language-leaders!

You might have noticed in a production environment that the `weather_HassGetWeather` intent doesn't really work (it doesn't provide the weather condition) and that the `person_HassGetState` intent is really hit and miss.

The reason behind this is that both intents [use](https://github.com/home-assistant/intents/blob/5c76e668ceccb233a604567c0dd9d96cfdd0cd9a/responses/en/HassGetWeather.yaml#L24C134-L24C134) the [state](https://github.com/home-assistant/intents/blob/5c76e668ceccb233a604567c0dd9d96cfdd0cd9a/responses/en/HassGetState.yaml#L70) value for making decisions upon which to build the response, but the state [is translated](https://github.com/home-assistant/core/blob/fb615817b4ccb3a0dcfeaf0384a1ff29633170cd/homeassistant/components/conversation/default_agent.py#L335) before being used in the intent response. This is by design, as you want to say that a battery "isn't on, but off" in the same language in which the question was received.

Since we're less than 36h away from the [Chapter 5 stream](https://www.youtube.com/watch?v=djEkgoS5dDQ), I was going to propose we fix the sentences for `weather` and `person` state in the short time we have until then, so that people can enjoy the functionality as soon as possible.

To prevent breaking changes, I've made [a core PR](https://github.com/home-assistant/core/pull/105524) that introduces both the translated and untranslated state as state attributes, thus preserving the current translated state behavior. The PR is still under review, though, so I wouldn't recommend that you fix the sentences/responses/tests directly on the [`main` branch](https://github.com/home-assistant/intents/tree/main) of the intents repo. Instead, I've created an [`untranslated-state` branch](https://github.com/home-assistant/intents/tree/untranslated-state), upon which I've [opened a PR](https://github.com/home-assistant/intents/pull/1767) to fix the English sentences. ~~This PR is also pending review and I'd recommend you subscribe to it to get notifications in case anything changes.~~ The PR has been approved and merged, so it's a good example of what needs to be done

I kindly ask you to do the same for the languages you are leading as I did for English:
- fetch the `untranslated-branch` in your fork: `git fetch upstream untranslated-state`, assuming the remote pointing to [`home-assistant/intents`](https://github.com/home-assistant/intents) is called `upstream`
- check out the branch: `git checkout untranslated-state`
- create a feature branch from that branch: `git branch xx-untranslated-changes` where `xx` could be a language code. Feel free to name your branch however you want, though
- fix the sentences according to [this model](https://github.com/home-assistant/intents/pull/1767/files)
- commit and push your changes to your fork
- open a PR **against the `untranslated-state` branch** by using a URL similar to this https://github.com/home-assistant/intents/compare/untranslated-state...your-name:intents:xx-untranslated-state (you **need to edit the URL** to make it work)
- get reviews and merge the PR in the `untranslated-state` branch as soon as you ~~see #1767 being merged (changes may still occur)~~ can

Once https://github.com/home-assistant/core/pull/105524 gets merged into `core`, we can also merge the new branch into the intents repo and create a new release. That way, we will sync our effort and not introduce breaking changes.

**Comment by mib1185** [comment]
will do so today evening (CET) for German language -> Done #1777

**Reply by mib1185** [reply]
`person_HassGetState` is not implemented in German, yet - but we have an open PR #1657.
But even I would merge this PR now, [untranslated-state](https://github.com/home-assistant/intents/tree/untranslated-state) is not up-to-date than 🤔

**Reply by tetele** [reply]
It won't work if you merge it to `main` anyway, so you could rebase it to the feature branch and move the PR there.

**Reply by mib1185** [reply]
ok ... just to be sure, so we need to rebase #1657 onto [untranslated-state](https://github.com/home-assistant/intents/tree/untranslated-state), than going to merge into [untranslated-state](https://github.com/home-assistant/intents/tree/untranslated-state), afterwards go ahead with adopting #1767 for German language?

**Reply by mib1185** [reply]
rebase and merge is done, now going ahead with adopting #1767 for German language

**Reply by tetele** [reply]
That's the game plan, yes. I mean... the point is to have these features available on the `untranslated-state` branch.

**Reply by mib1185** [reply]
Done #1777

**Reply by tetele** [reply]
Thanks! When the checks are done, you can merge it to the feature branch

**Comment by Uriziel01** [comment]
@tetele Do you want us to do it ASAP or are you expecting any further changes to the #1767 and we should wait for it to be finalized and merged?

**Reply by tetele** [reply]
I'm not planning any changes, but I would like to get @synesthesiam's review before merging. I think the fastest approach is to translate the current changes and subscribe to any reviews for #1767 to see if there will be any updates.

I haven't got Mike's confirmation on pushing a new intents release, so I can't guarantee we will have a new release tomorrow, but I hope we will be able to. Also, keep in mind you don't **have** to make these changes, it was just that I thought the stream would drive some adoption and testing and it would have been better if stuff worked :D

**Reply by tetele** [reply]
[There's your answer](https://github.com/home-assistant/intents/discussions/1768#discussioncomment-7835433). No further changes, your honor! :D

**Reply by Uriziel01** [reply]
@tetele PR is ready https://github.com/home-assistant/intents/pull/1779 I was also forced to do whole lot more than just what was in scope if this discussion (details in the PR description).

Sorry for all the people that got notified that a new PR with their language label was created, when PR was just in draft mode I've had it target branch set to `main` for a moment a GitHub decided this is the perfect moment to add all of your labels, so... yeah 🥳

**Reply by Uriziel01** [reply]
Hi @tetele, any ETA on when you'll fix the the "small hiccup" in https://github.com/home-assistant/core/pull/105524 so we can merge the https://github.com/home-assistant/intents/tree/untranslated-state back into main? Thank you.

**Reply by tetele** [reply]
You'll need to ask @synesthesiam about that. Apparently, it's not necessarily the preferred approach, although usable. Mike said that he would make the changes himself (without providing a timeline), but I haven't seen him online in a while. The initial idea was for this change to be included in Chapter 5, but no release was made specifically for it.

Anyway, long story short, no, I don't have an ETA. The `untranslated-state` merge depends on the fix in `core` and I have no idea when that will be merged.

**Comment by davefx** [comment]
Spanish changes completed, waiting for https://github.com/home-assistant/core/pull/105524.

**Comment by TheFes** [comment]
NL done as well https://github.com/home-assistant/intents/pull/1771

**Comment by synesthesiam** [comment]
PR's are approved and waiting to be merged. One small hiccup in the HA core PR, but it will be fixed soon.

Please proceed with the changes :)

**Comment by jmceara** [comment]
PT-BR done [#1786](https://github.com/home-assistant/intents/pull/1786)

**Reply by tetele** [reply]
@jmceara bummer, there's no language leader for Brazilian Portuguese. Wanna fill in those shoes? It's not a hard job, but we would hate to see you ignore it.

**Reply by jmceara** [reply]
Let's go! :)

**Reply by tetele** [reply]
@synesthesiam please see #1786 (which accounts as at least a few PRs) and anoint @jmceara as knight of the language leaders if you so see fit.

**Reply by llluis** [reply]
I was about to create a PR for pt-br translations when I found the existing ones and this topic. So, count me in to help @jmceara. 
(I would gladly be a co-leader with you).

**Comment by duhow** [comment]
Catalan [CA] done at #1787

**Reply by tetele** [reply]
I've added a review and asked for a small change. It's true it was not clear in English which is the state and which is the translation

**Comment by fadamsen** [comment]
I will try to get Danish (DA) done tonight.
